src/librustdoc/clean/inline.rs

```diff
@@ -387,10 +396,8 @@ pub(crate) fn build_impl(
                 return;
             }
 
-            if let Some(stab) = tcx.lookup_stability(did) {
-                if stab.is_unstable() && stab.feature == sym::rustc_private {
-                    return;
-                }
+            if !document_compiler_internal && let Some(stab) = tcx.lookup_stability(did) && stab.is_unstable() && stab.feature == sym::rustc_private {
```

Comment:
This is really a nit, not a blocker: this code is very similar in the three cases. Couldn't you write a closure/function to handle this code instead of repeating it 3 times?